pkg/cmd/infra/router/template.go

```diff
@@ -287,6 +294,77 @@ func parseCaptureHeaders(in string) ([]templateplugin.CaptureHTTPHeader, error)
 	return captureHeaders, nil
 }
 
+func sanitizeHeaderValue(headerValue string) string {
+	santizeHeaderValue := strings.ReplaceAll(headerValue, "'", `'\''`)
+	return "'" + santizeHeaderValue + "'"
+}
+
+func parseHeadersToBeSetOrDeleted(in string) ([]templateplugin.HTTPHeader, error) {
+	var captureHeaders []templateplugin.HTTPHeader
+	var capture templateplugin.HTTPHeader
+	/*encodedIn, err := b64.StdEncoding.DecodeString(in)
+	if err != nil {
+		return captureHeaders, fmt.Errorf("failed to decode base64 encoding1: %v", encodedIn)
+	}*/
+	if len(in) > 0 {
+		for _, header := range strings.Split(in, ",") {
+			parts := strings.Split(header, ":")
+			if len(parts) < 2 {
+				return captureHeaders, fmt.Errorf("invalid HTTP header input specification: %v", header)
+			}
+			if len(parts) == 3 {
+				headerName, err := b64.StdEncoding.DecodeString(parts[0])
+				if err != nil {
+					return captureHeaders, fmt.Errorf("failed to decode base64 encoding2: %v", headerName)
+				}
+				// RFC 2616, section 4.2, states that the header name
+				// must be a valid token.
+				if !validTokenRE.MatchString(string(headerName)) {
+					return captureHeaders, fmt.Errorf("invalid HTTP header name: %v", headerName)
+				}
+				headerValue, err := b64.StdEncoding.DecodeString(parts[1])
+				if err != nil {
+					return captureHeaders, fmt.Errorf("failed to decode base64 encoding3: %v", headerValue)
+				}
+				sanitizedHeaderValue := sanitizeHeaderValue(string(headerValue))
+				action, err := b64.StdEncoding.DecodeString(parts[2])
```

Comment:
It would be best to make sure the `action` is either set or delete, just in case someone misconfigured.

Reply:
I am not sure because we allow both actions to be added at the same time. Also both actions have different way to provide a value.

Reply:
So, a test for the actual value Set or Delete, not a typo like Seet or Del.